### PR TITLE
[Manifest] Use kustomize native image transformer to override image

### PR DESCRIPTION
**Pull request by Bobgy**
/assign @rmgogogo 
/cc @IronPan 

Part of https://github.com/kubeflow/pipelines/issues/2884
> Images listed in kustomization.yaml: required

### CODE REVIEW

File: test/deploy-pipeline-lite.sh [hunk 1]
```diff
@@ -47,9 +49,19 @@ if [ -z "$KFP_DEPLOY_RELEASE" ]; then
   popd
 
   pushd ${KFP_MANIFEST_DIR}/dev
-  echo imagePrefix=${GCR_IMAGE_BASE_DIR} > params.env
-  echo imageTag=${GCR_IMAGE_TAG} >> params.env  
-  cat params.env
+  # This is the recommended approach to do this.
+  # reference: https://github.com/kubernetes-sigs/kustomize/blob/master/docs/eschewedFeatures.md#build-time-side-effects-from-cli-args-or-env-variables
```
Comment (rmgogogo) [inline test/deploy-pipeline-lite.sh:53]: I'm wondering whether the change increased the complexity actually.
Feel free to go, just feel "kustomize edit set image" didn't well solve the problem

**Comment by rmgogogo** [comment]
would you also change the README.md for a sample installation?
I assume it's better for a quick installation, user doesn't need to edit the image tag.

File: test/deploy-pipeline-lite.sh (hunk 1 above)
Comment (Bobgy) [reply test/deploy-pipeline-lite.sh:53]: Complexity for test scripts seem trivial to me. The important part is it left the flexibility of overriding images to users by kustomize.

**Comment by Bobgy** [comment]
> would you also change the README.md for a sample installation?
> I assume it's better for a quick installation, user doesn't need to edit the image tag.

@rmgogogo I don't quite follow. I think there isn't any user-visible change except that images can be overriden by kustomize again. Why would I need to update README.md for sample install?

**Comment by rmgogogo** [comment]
/lgtm
/approve

as synced offline, user still can quickly install KFP without touch "kubectl edit images" if don't want to customize images.

**Comment by Bobgy** [comment]
/retest
presubmit fixed

**Pull request merged**